--- kmobile/devices/gnokii/gnokii_mobile.cpp	2006-08-06 12:35:00.000000000 +0200
+++ kmobile/devices/gnokii/gnokii_mobile.cpp	2006-08-06 12:49:54.000000000 +0200
@@ -741,20 +741,20 @@
   case GN_CALNOTE_NEVER:
 		break;
   case GN_CALNOTE_DAILY:
-		event.recurrence()->setDaily(1,-1);
+		event.recurrence()->setDaily(1);
 		break;
   case GN_CALNOTE_WEEKLY:
   case GN_CALNOTE_2WEEKLY:
-		event.recurrence()->setDaily( 7 + (entry.recurrence==GN_CALNOTE_2WEEKLY ? 7:0) , -1);
+		event.recurrence()->setDaily( 7 + (entry.recurrence==GN_CALNOTE_2WEEKLY ? 7:0));
 		break;
   case GN_CALNOTE_MONTHLY:
-		event.recurrence()->setMonthly(KCal::Recurrence::rMonthlyPos, 1, -1);
+		event.recurrence()->setMonthly(KCal::Recurrence::rMonthlyPos);
 		break;
   case GN_CALNOTE_YEARLY:
-		event.recurrence()->setYearly(KCal::Recurrence::rYearlyPos, 1, -1);
+		event.recurrence()->setYearly(KCal::Recurrence::rYearlyPos);
 		break;
   default: // hourly
-		event.recurrence()->setHourly(entry.recurrence, -1);
+		event.recurrence()->setHourly(entry.recurrence);
 		break;
   }
 
