--- ktorrent/src/Makefile.am	2005-06-04 16:05:29.000000000 -0700
+++ ktorrent/src/Makefile.am	2005-06-04 16:00:30.000000000 -0700
@@ -7,13 +7,13 @@
 
 
 # set the include path for X, qt and KDE
-INCLUDES = -I$(top_srcdir)/. $(all_includes)
+INCLUDES = -I$(top_srcdir)/ktorrent/. $(all_includes)
 
 # the library search path. 
 ktorrent_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 # the libraries to link against.
-ktorrent_LDADD = $(top_builddir)/libtorrent/libtorrent.la $(LIB_KFILE) \
+ktorrent_LDADD = $(top_builddir)/ktorrent/libtorrent/libtorrent.la $(LIB_KFILE) \
 	$(LIB_KDEPRINT) $(LIB_KHTML)
 
 # which sources should be compiled for ktorrent
