*** trunk/extragear/multimedia/kaffeine/src/kaffeine.cpp	2006-12-17 05:54:01.000000000 +0100
--- trunk/extragear/multimedia/kaffeine/src/kaffeine.cpp	2006-12-17 09:33:46.000000000 +0100
***************
*** 55,60 ****
--- 55,63 ----
  #include <kinputdialog.h>
  #include <kcmdlineargs.h>
  #include <dcopclient.h>
+ #include <kapp.h>
+ #include <kapplication.h>
+ #include <kdeversion.h>
  
  #include <qdir.h>
  #include <qlayout.h>
