--- kplato/kptproject.h.old	2006-07-16 17:55:41.000000000 +0200
+++ kplato/kptproject.h	2006-07-16 17:55:51.000000000 +0200
@@ -179,7 +179,7 @@
     /// Insert the node with identity id
     virtual void insertId(const QString &id, const Node *node);
     /// Register node. The nodes id must be unique and non-empty.
-    bool Project::registerNodeId(Node *node);
+    bool registerNodeId(Node *node);
     /// Create a unique id.
     QString uniqueNodeId(int seed=1);
     
