--- KDE/kdepim/libkcal/libical/src/libical/icaltimezone.c.orig	2005-06-26 14:06:51.847950520 +0200
+++ KDE/kdepim/libkcal/libical/src/libical/icaltimezone.c	2005-06-26 14:07:01.166533880 +0200
@@ -45,7 +45,7 @@
 #endif
 
 /** This is the toplevel directory where the timezone data is installed in. */
-#define ZONEINFO_DIRECTORY	PACKAGE_DATA_DIR "/zoneinfo"
+#define ZONEINFO_DIRECTORY	PACKAGE_DATA_DIR + "/zoneinfo"
 
 /** The prefix we use to uniquely identify TZIDs. */
 #define TZID_PREFIX		"/softwarestudio.org/"
